app/src/main/jni/interpreter.c

```diff
@@ -1,5 +1,44 @@
 #include "interpreter.h"
 
+/* Verbosely execute Python code.
+ * Exceptions are print to Android log using LOGI macro.
+ * Returns 0 on success or -1 if an exception was raised.
+ */
+int Verbose_PyRun_SimpleString(const char *code) {
+  // Used for globals and locals
+  PyObject *module = PyImport_AddModule("__main__");
+  PyObject *d = PyDict_New();
+  if (module == NULL)
+    return -1;
```

Comment:
Do you need to set errno (or the Python equivalent)? 

Also, at a minimum, it seems to make sense to have a comment that explains when you expect this situation to be true...